### PR TITLE
Fix spurious Windows errors with switch_features_rerun.

### PR DESCRIPTION
**Pull request by ehuss**
Fix a flaky test on Windows which fails with "Access is denied.".

Previous errors:
https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/17766698
https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/18040797
https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/20472056
https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/18746372
https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/18559776
https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/19659571
https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/18956349
https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/18183191
https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/20334485
https://ci.appveyor.com/project/rust-lang-libs/cargo/builds/18554058

### CODE REVIEW

File: tests/testsuite/support/mod.rs [hunk 1]
```diff
@@ -399,6 +399,27 @@ impl Project {
...
+    /// The action may fail (with errors like Access is denied), or
+    /// it may succeed, but future attempts to use the same filename
+    /// will fail with "Already Exists".
+    ///
+    /// If you have a test that needs to do `cargo run` multiple
+    /// times, you should instead use `cargo build` and use this
+    /// method to run the executable. Each time you call this,
+    /// use a new name for `dst`.
+    /// See https://github.com/rust-lang/cargo/issues/5481
+    pub fn safely_rename_run(&self, src: &str, dst: &str) -> Execs {
```
Comment (dwijnand) [inline tests/testsuite/support/mod.rs]: 👨‍🎨 maybe just called `run` and with a name automatically derived from src?

Comment (ehuss) [reply tests/testsuite/support/mod.rs]: Yea, it's a bit wordy. However, I don't want to infer that this is a normal way of running an executable.  I dropped the "safely" bit.

To derive the name, it would need to maintain state somewhere to keep the name unique.  `Project` isn't generally mutable, so it seemed a bit awkward. So I'm unsure if there's a clean way to do that.

Comment (dwijnand) [reply tests/testsuite/support/mod.rs]: > Yea, it's a bit wordy. However, I don't want to infer that this is a normal way of running an executable. I dropped the "safely" bit.

It avoids a pitfall on Windows, under certain circumstances.  If there's no penalty hit I wouldn't mind people using it as the normal way of running an executable.  WDYT?

> To derive the name, it would need to maintain state somewhere to keep the name unique. `Project` isn't generally mutable, so it seemed a bit awkward. So I'm unsure if there's a clean way to do that.

I'd just set up another one of these as a run naming counter:

https://github.com/rust-lang/cargo/blob/87dd6c279ab8b499118677733958e29362793d6e/tests/testsuite/support/paths.rs#L12-L14

**Comment by alexcrichton** [comment]
@bors: r+

Thanks so much for tracking this down!

File: tests/testsuite/support/mod.rs (hunk 1 above)
Comment (ehuss) [reply tests/testsuite/support/mod.rs]: I would prefer if normal execution goes through `cargo run`. It's a somewhat different codepath which has had some issues in the past, so I think it's good for it to be fully exercised.  `cargo run` is also a bit more convenient to use.

Comment (dwijnand) [reply tests/testsuite/support/mod.rs]: Ah, I see.
